src/types/module.rs

```diff
@@ -189,57 +274,69 @@ impl PyModule {
         self.add(name, function)
     }
 
-    /// Add a submodule to a module.
+    /// Adds a submodule to a module.
     ///
-    /// Use this together with `#[pymodule]` and [wrap_pymodule!].
+    /// This is especially useful for creating module hierarchies.
```

Comment:
Well it does create a module hierarchy, but it doesn't (currently) act like a package - which is why I avoided using the word "package". In hindsight that's probably a lie by omission.

But it acting like a package is probably what people *expect* when they add submodules. What are your thoughts on adding something like this below it?

"Note that pyo3 does not yet support using modules as packages; see https://github.com/PyO3/pyo3/issues/1517#issuecomment-808664021 for a hack to make that work"

Reply:
I'm not sure it's actually *possible* to support it other than with what I posted there. As far as CPython is concerned, every module is a file, and since the compiled extension is also a file it is also treated as a single module (e.g. only one `PyInit_modname` function is used).

At the time I was looking but couldn't easily find C extensions that create a package hierarchy; I'm guessing that with C, the desire to write whole sets of modules is not as big as with Rust :D  With C, it is of course also more straightforward to provide lots of small `.so` libs - for Rust they'd have to be in separate crates.

Reply:
What about this?
```
Adds a submodule to a module.

This is especially useful for creating module hierarchies.

Note that this doesn’t define a package, so this won’t allow Python code to directly import submodules like from my_module import submodule. For more information, see #759 and #1517.
```

Reply:
Sounds good for now.